### PR TITLE
feat: Add Todoist provider

### PR DESCRIPTION
**Pull request by elpnt**
## ☕️ Reasoning

Adds [Todoist](https://todoist.com/home) to the available providers.

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

## 🎥 Demo video

Run `pnpm dev:app` and it works fine

https://user-images.githubusercontent.com/39664774/187518215-c67c4816-ee78-4ab8-8042-21fc4cb72248.mp4

### CODE REVIEW

File: packages/next-auth/src/providers/todoist.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            client_id: options.clientId,
+            client_secret: options.clientSecret,
+            code: params.code,
+          }),
+        })
+
+        const tokens = await res.json()
+        return { tokens }
+      },
+    },
```
Comment (ThangHuuVu) [inline packages/next-auth/src/providers/todoist.ts:74]: Can you try the client option: `client_secret_post`? 
You won't need to do the request here, `openid-client` will take the option and do it for you
```
    token: "https://todoist.com/oauth/access_token",
    client: {
      token_endpoint_auth_method: "client_secret_post",
    },
```

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]

**Comment by elpnt** [comment]
I rewrote the token fetch process and it works fine. I also removed unused field type definitions from `TodoistProfile`

**Comment by ThangHuuVu** [comment]
Thanks for sending this @elpnt 🙌, could you add the provider to the list in `2_bug_provider.yml`? That's the last required change in this PR!

**Review by ThangHuuVu — APPROVED** [review]
LGTM!

**Pull request merged**